File: leetcode_go/problems/0217_contains_duplicate/contains_duplicate_test.go
package problem0217

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

type input struct {
	nums []int
}

type question struct {
	input  input
	output bool
}

func Test_contains_duplicate(t *testing.T) {
	questions := map[string]question{
		"1": {
			input: input{
				nums: []int{1, 2, 3, 1},
			},
			output: true,
		},
		"2": {
			input: input{
				nums: []int{1, 2, 3, 4},
			},
			output: false,
		},
		"3": {
			input: input{
				nums: []int{1, 1, 1, 3, 3, 4, 3, 2, 4, 2},
			},
			output: true,
		},
	}

	for name, question := range questions {
		t.Run(name, func(t *testing.T) {
			result := containsDuplicate(question.input.nums)
			assert.Equal(t, question.output, result)
		})
	}
}

File: leetcode_js/tests/containsDuplicate.test.js
const { containsDuplicate } = require('../src/containsDuplicate');

test('1', () => {
  const nums = [1, 2, 3, 1];
  const output = true;

  expect(containsDuplicate(nums)).toEqual(output);
});

test('2', () => {
  const nums = [1, 2, 3, 4];
  const output = false;

  expect(containsDuplicate(nums)).toEqual(output);
});

test('3', () => {
  const nums = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2];
  const output = true;

  expect(containsDuplicate(nums)).toEqual(output);
});
